opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+/**
+ * Tracer is the entry point API between instrumentation code and the tracing implementation.
+ */
+public interface Tracer {
+
+  /**
+   * Starts a new trace and creates a new root span.
+   *
+   * <p>This method should be used by services that are instrumented for tracing but did not receive
+   * trace ID from upstream request.
+   *
+   * @param operationName the service's own name for the end-point that received the request
+   * represented by this trace and span. The domain of names must be limited, e.g. do not use UUIDs
+   * or entity IDs or timestamps as part of the name.
+   */
+  Span startTrace(String operationName);
+
+  /**
+   * Joins a trace started elsewhere and creates a new span as a child of the given parent trace
+   * context.
+   *
+   * <p>This method should be used by services that receive tracing info from upstream.
+   *
+   * @param operationName the service's own name for the end-point that received the request
+   * represented by this trace and span. The domain of names must be limited, e.g. do not use UUIDs
+   * or entity IDs or timestamps as part of the name.
+   * @param parent Trace Context of a client span started elsewhere and whose trace we're joining.
+   */
+  Span joinTrace(String operationName, TraceContext parent);
```

Comment:
no, retaining the same span_id can be easily done as internal impl of zipkin when you do join_trace.

StartSpanWithContext was primarily added to support "debug" traces, where the instrumentation decides that a trace should be sampled no matter what, creates a new context, sets a debug attribute on the context, and then creates a span with that context.